src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -605,18 +715,27 @@ private static bool HasDuplicateInterfaces(NamedTypeSymbol type, ConsList<Symbol
             Compilation currentCompilation,
             ArrayBuilder<TypeParameterDiagnosticInfo> diagnosticsBuilder,
             ref ArrayBuilder<TypeParameterDiagnosticInfo> useSiteDiagnosticsBuilder,
+            ref ArrayBuilder<DiagnosticInfo> tupleDiagnosticsBuilder,
             BitVector skipParameters = default(BitVector))
         {
-            return CheckConstraints(
-                method,
-                conversions,
-                method.TypeSubstitution,
-                ((MethodSymbol)method.OriginalDefinition).TypeParameters,
-                method.TypeArguments,
-                currentCompilation,
-                diagnosticsBuilder,
-                ref useSiteDiagnosticsBuilder,
-                skipParameters);
+            bool result = CheckConstraints(
+                            method,
+                            conversions,
+                            method.TypeSubstitution,
+                            ((MethodSymbol)method.OriginalDefinition).TypeParameters,
+                            method.TypeArguments,
+                            currentCompilation,
+                            diagnosticsBuilder,
+                            ref useSiteDiagnosticsBuilder,
+                            skipParameters);
+
+            foreach (var parameter in method.Parameters)
+            {
+                result &= CheckTupleNamesAgainstBase(parameter.Type, ref tupleDiagnosticsBuilder);
```

Comment:
>CheckTupleNamesAgainstBase(parameter.Type, ref tupleDiagnosticsBuilder); [](start = 26, length = 72)

It feels like this is going to catch the problem only at the top level. We probably should also dig into type arguments, arrays, pointers, etc.